redis/pool.go

```diff
@@ -299,34 +299,61 @@ func (p *Pool) lazyInit() {
 	p.mu.Unlock()
 }
 
-// get prunes stale connections and returns a connection from the idle list or
-// creates a new connection.
-func (p *Pool) get(ctx context.Context) (*poolConn, error) {
+// waitVacantConn waits while at least one vacant connection appears in the pool.
+// If `ctx` expires before that, an error is returned.
+//
+// waitVacantConn returns:
+// - (as its first return value) the time it spent waiting, if there were no vacant connection in the pool right away
+// or zero otherwise
+// - (as its second return value) error, if any
```

Comment:
It's not idiomatic to describe a return like this, see some of the core libraries for examples e.g.
https://golang.org/pkg/io/#Copy

Reply:
Changed `func` signature and its description a bit.